src/clients/cli.rs

```diff
@@ -167,12 +192,24 @@ impl Docker for Cli {
             .spawn()
             .expect("Failed to execute docker command");
     }
+    /// This transparently returns the ip address of the container `id`,
+    /// as reported by `docker inspect`.
+    /// Note: The user needs to make sure that the physical or virtual node,
+    /// e.g. wherever the orchestrating code is running, is able to route
+    /// to this IP address. In general, this is achieved by relying on the
+    /// default docker bridge network.
+    fn ip_address(&self, id: &str) -> String {
+        let info = self.get_container_info(id);
+        info.network_settings.ip_address
```

Comment:
if container is connected to a network that is different from the tests' one, you will not be able to use this IP.

This is why in testcontainers-java we always connect to the gateway's IP